src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPKey.MLDsa.cs

```diff
@@ -0,0 +1,179 @@
+        [Experimental(Experimentals.PostQuantumCryptographyDiagId)]
+        internal static partial class EvpPKeyMLDsa
+        {
+            internal static string? MLDsa44 { get; }
+            internal static string? MLDsa65 { get; }
+            internal static string? MLDsa87 { get; }
+
+            static EvpPKeyMLDsa()
+            {
+                CryptoInitializer.Initialize();
+
+                // Do not use property initializers for these because we need to ensure CryptoInitializer.Initialize
+                // is called first. Property initializers happen before cctors, so instead set the property after the
+                // initializer is run.
+                MLDsa44 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa44.Name);
+                MLDsa65 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa65.Name);
+                MLDsa87 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa87.Name);
+            }
+
+            [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+            private static partial int CryptoNative_IsSignatureAlgorithmAvailable(string algorithm);
+
+            private static string? IsSignatureAlgorithmAvailable(string algorithm)
+            {
+                const int Available = 1;
+                const int NotAvailable = 0;
+
+                int ret = CryptoNative_IsSignatureAlgorithmAvailable(algorithm);
+                return ret switch
+                {
+                    Available => algorithm,
+                    NotAvailable => null,
+                    int other => throw Fail(other),
+                };
+
+                static CryptographicException Fail(int result)
+                {
+                    Debug.Fail($"Unexpected result {result} from {nameof(CryptoNative_IsSignatureAlgorithmAvailable)}");
+                    return new CryptographicException();
+                }
+            }
+
+            [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+            private static partial SafeEvpPKeyHandle? CryptoNative_MLDsaGenerateKey(string keyType, ReadOnlySpan<byte> seed, int seedLength);
+
+            public static SafeEvpPKeyHandle MLDsaGenerateKey(string algorithmName, ReadOnlySpan<byte> seed)
+            {
+                SafeEvpPKeyHandle? handle = CryptoNative_MLDsaGenerateKey(algorithmName, seed, seed.Length);
+
+                if (handle == null || handle.IsInvalid)
+                {
+                    throw Interop.Crypto.CreateOpenSslCryptographicException();
+                }
+
+                return handle;
+            }
+
+            [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+            private static partial SafeEvpPKeyHandle? CryptoNative_MLDsaImportSecretKey(string keyType, ReadOnlySpan<byte> sk, int skLength);
+
+            public static SafeEvpPKeyHandle MLDsaImportSecretKey(string algorithmName, ReadOnlySpan<byte> sk)
```

Comment:
I know that the C# language rules say that a public method inside an internal class is really just `internal`; but I feel like "if it's really `internal`, say that"... so my preference is that all of the `public` methods in Interop.Crypto be declared `internal` instead.